src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -21,7 +21,7 @@ internal struct SyntaxTreeInfo
 
             public SyntaxTreeInfo(string filePath, ParseOptions options, ValueSource<TextAndVersion> textSource, Encoding encoding, int length)
             {
-                FilePath = filePath;
+                FilePath = filePath ?? string.Empty;
```

Comment:
Dumb question.  Why do we need this part?  If they asked for a null path, why not just keep the nll path?